$OpenBSD: patch-src_openvpn_options_c,v 1.1 2013/04/20 16:22:55 sthen Exp $

the rtable support in the 2.31 update is suspect; however it is does
also not appear to be particularly useful, seems to only do the same thing
as 'route -T ## exec openvpn' .. possible candidate for removal, or reworking
to do something more clever.

--- src/openvpn/options.c.orig	Fri Mar 22 18:02:30 2013
+++ src/openvpn/options.c	Mon Apr  1 17:55:36 2013
@@ -304,6 +304,7 @@ static const char usage_message[] =
   "                  can be matched in policy routing and packetfilter rules.\n"
 #endif
   "--txqueuelen n  : Set the tun/tap TX queue length to n (Linux only).\n"
+  "--rtable n      : Set the routing table (default=0, OpenBSD only)\n"
 #ifdef ENABLE_MEMSTATS
   "--memstats file : Write live usage stats to memory mapped binary file.\n"
 #endif
@@ -1502,6 +1503,7 @@ show_settings (const struct options *o)
 #endif
   SHOW_INT (rcvbuf);
   SHOW_INT (sndbuf);
+  SHOW_INT (rtable);
 #if defined(TARGET_LINUX) && HAVE_DECL_SO_MARK
   SHOW_INT (mark);
 #endif
@@ -4782,6 +4784,11 @@ add_option (struct options *options,
     {
       VERIFY_PERMISSION (OPT_P_SOCKBUF);
       options->sndbuf = positive_atoi (p[1]);
+    }
+  else if (streq (p[0], "rtable") && p[1])
+    {
+      VERIFY_PERMISSION (OPT_P_SOCKFLAGS);
+      options->rtable = positive_atoi (p[1]);
     }
   else if (streq (p[0], "mark") && p[1])
     {
